src/main/java/bio/terra/service/snapshot/flight/create/NotifyUserOfSnapshotCreationStep.java

```diff
@@ -0,0 +1,31 @@
+package bio.terra.service.snapshot.flight.create;
+
+import bio.terra.common.iam.AuthenticatedUserRequest;
+import bio.terra.service.job.DefaultUndoStep;
+import bio.terra.service.snapshotbuilder.SnapshotBuilderService;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+import java.util.UUID;
+
+public class NotifyUserOfSnapshotCreationStep extends DefaultUndoStep {
```

Comment:
What is this class for and what is the DefaultUndoStep class?

Reply:
This is a step, which is part of a flight. These are concepts in the stairway library, which TDR uses to perform long running operations that are undo-able. See https://github.com/DataBiosphere/stairway?tab=readme-ov-file#flights-and-steps for more information about flights and steps.

This step has no undo operation (since we can't un-send an email) so I use `DefaultUndoStep` where the undo does nothing. If I didn't use that base class, I'd have to provide my own empty `undoStep()` method to satisfy the `Step` interface.